tests/unit/test-cmd/test.run.js

```diff
@@ -326,101 +326,91 @@ describe('run', () => {
            sourceDir: '/fake/source/dir',
          },
        );
-     }
-  );
+     });
 
   describe('profile-create-new option', () => {
     beforeEach(() => {
       sinon.stub(fs, 'mkdir');
+      sinon.stub(fs, 'existsSync');
     });
 
     afterEach(() => {
       fs.mkdir.restore();
+      fs.existsSync.restore();
     });
 
-    it('creates dir when firefox profile doesn\'t exist',
-       async () => {
-         sinon.stub(fs, 'existsSync').returns(false);
-         const firefoxProfile = '/pretend/path/to/Firefox/profile';
-         const cmd = prepareRun();
-
-         await cmd.run({
-           firefoxProfile,
+    const fakeFirefoxProfile = '/pretend/path/to/Firefox/profile';
+    const fakeChromiumProfile = '/pretend/path/to/Chromium/profile';
+
+    async function testCreateProfileIfMissing(
+      existsSyncReturnValue,
+      runParams,
+      expectedProfile
+    ) {
+      fs.existsSync.returns(existsSyncReturnValue);
+      const cmd = prepareRun();
+
+      await cmd.run(runParams);
+
+      if (existsSyncReturnValue) {
+        sinon.assert.notCalled(fs.mkdir);
+      } else {
+        sinon.assert.calledWith(fs.mkdir, expectedProfile);
+      }
+
+      if (expectedProfile === fakeFirefoxProfile) {
+        sinon.assert.calledOnce(desktopRunnerStub);
+        const firefoxProfile = desktopRunnerStub
+          .firstCall
+          .args[0]
+          .profilePath;
+        assert.equal(firefoxProfile, fakeFirefoxProfile,
+                     'Got the expected firefoxProfile option');
+      } else {
+        sinon.assert.calledOnce(chromiumRunnerStub);
+
+        const {chromiumProfile} = chromiumRunnerStub.firstCall.args[0];
+        assert.equal(chromiumProfile, fakeChromiumProfile,
+                     'Got the expected chromiumProfile option');
+      }
+    }
+
+    it('creates dir when firefox profile does not exist',
+       async () => testCreateProfileIfMissing(
+         false, {
+           firefoxProfile: fakeFirefoxProfile,
            profileCreateIfMissing: true,
-         });
-
-         sinon.assert.calledWith(fs.mkdir, firefoxProfile);
-
-         fs.existsSync.restore();
-       }
+         }, fakeFirefoxProfile
+       )
```

Comment:
Nitpick:  would you mind to move this `)` to the previous line and keep the parameters on separate lines?
I mean something like:
```js
testCreateProfileIfMissing(
  false,
  {
    ...
  },
  fakeFirefoxProfile);
```

as it is more or less the convention we do use at the moment, same for other calls that follow.